jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java

```diff
@@ -156,9 +164,11 @@ private RegistryAuthenticator(
    * @param credential the credential used to authenticate
    * @return an {@code Authorization} authenticating the pull
    * @throws RegistryAuthenticationFailedException if authentication fails
+   * @throws RegistryCredentialsNotSentException if authentication is failed and credentials were
```

Comment:
This seems an odd name.  I think something about _insecure_ matters here.  Maybe it should be a code on `RegistryAuthenticationFailedException`?

Reply:
Yeah, I don't really like it's throwing an additional exception, but `RegistryCredentialsNotSentException` is an existing exception that we introduced specifically to identify the case where Jib got 401 (unauthorized) over plain-HTTP when `Authorization:` was not sent and then catch this to show a nice message that the user may want to use `-DsendCredentialsOverHttp`. There is code at an upper level that catches this exception. There may be better ways to restructure things for this, but I don't have a concrete idea now that will make it look better. :/